### PR TITLE
Onchain reward credentials

### PR DESCRIPTION
**Pull request by motechFR**
### WHAT

Reward credentials
- Issue onchain and offchain reward credentials

On / offchain
- A single issued credential can reference an on and offchain record
- Better testing, deduplication measures in place

Gnosis safe
- Fixed indexing method for gnosis safe transactions, so we only check multicall transaction logs which call EAS contract
- Index pending transactions, and delete them from DB once complete

Ceramic
- Read credential data from offchain data sources

### WHY

We needed reward credentials to work alongside proposal credentials to launch this, as well as better Gnosis safe handling

### CODE REVIEW

File: charmClient/apis/credentialsApi.ts [hunk 1]
```diff
@@ -34,7 +34,15 @@ export class CredentialsApi {
     return http.POST(`/api/credentials/proposals/request-indexing`, data);
   }
 
-  requestPendingProposalCredentialGnosisSafeIndexing(data: GnosisSafeTransactionToIndex) {
-    return http.POST(`/api/credentials/proposals/request-gnosis-safe-indexing`, data);
+  requestPendingCredentialGnosisSafeIndexing(data: GnosisSafeTransactionToIndex) {
+    return http.POST(`/api/credentials/request-gnosis-safe-indexing`, data);
+  }
+
+  getIssuableRewardCredentials({ spaceId }: { spaceId: string }) {
```
Comment (mattcasey) [inline charmClient/apis/credentialsApi.ts:41]: please use hook format for retrieving api calls in the future

File: lib/credentials/indexGnosisSafeCredentialTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    // Mark this transaction as processed so we can debug it later
+    await prisma.pendingSafeTransaction.update({
+      where: { safeTxHash },
+      data: { processed: true, credentialContent: newValues }
+    });
+  }
+}
+
+// First
+// indexGnosisSafeCredentialTransaction({
```
Comment (mattcasey) [inline lib/credentials/indexGnosisSafeCredentialTransaction.ts:253]: remove this?

File: testing/setupDatabase.ts [hunk 3]
```diff
@@ -250,6 +250,9 @@ export async function generateUserAndSpace({
   };
 }
 
+/**
+ * @customPageId used for different reward and page ids to test edge cases where these ended up different
```
Comment (mattcasey) [inline testing/setupDatabase.ts:254]: :)

File: .github/actions/install/action.yml [hunk 4]
```diff
@@ -39,7 +39,7 @@ runs:
       uses: actions/cache/restore@v4
       with:
         path: ./node_modules
-        key: nodemodules-cache-${{ hashFiles('package-lock.json') }}-${{ hashFiles('patches/**.patch', 'prisma/schema.prisma') }}
+        key: nodemodules-cache-k-${{ hashFiles('package-lock.json') }}-${{ hashFiles('patches/**.patch', 'prisma/schema.prisma') }}
```
Comment (mattcasey) [inline .github/actions/install/action.yml:42]: remember to remove this

**Review by mattcasey — APPROVED** [review]
